gateway-js/src/index.ts

```diff
@@ -351,24 +430,41 @@ export class ApolloGateway implements GraphQLService {
       }
     }
 
-    await this.updateComposition();
-    if (
-      (isManagedConfig(this.config) || this.experimental_pollInterval) &&
-      !this.pollingTimer
-    ) {
-      this.pollServices();
-    }
+    const result = isStaticConfig(this.config)
+      ? this.loadStatic()
+      : await this.loadDynamic();
 
     const mode = isManagedConfig(this.config) ? 'managed' : 'unmanaged';
-
     this.logger.info(
       `Gateway successfully loaded schema.\n\t* Mode: ${mode}${
-        (this.apolloConfig && this.apolloConfig.graphId)
+        this.apolloConfig && this.apolloConfig.graphId
           ? `\n\t* Service: ${this.apolloConfig.graphId}@${this.apolloConfig.graphVariant}`
           : ''
       }`,
     );
 
+    return result;
+  }
+
+  private loadStatic() {
+    this.maybeWarnOnConflictingConfig();
+
+    return {
+      schema: this.schema!,
+      executor: this.executor,
+    };
+  }
+
+  private async loadDynamic() {
+    if (isRemoteConfig(this.config)) {
+      this.maybeWarnOnConflictingConfig();
+    }
+
+    await this.updateComposition();
```

Comment:
I wonder if it would be helpful to rename updateComposition and updateServiceDefinitions to contain "dynamic", to be more clear they are only part of the dynamic case? On the other hand maybe this is implicit in the concept of updating.